src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -250,9 +247,9 @@ void Gather::prepareParams() {
         specIdxAndAfterAxSizeB = specIndicesSize * afterAxisSizeInBytes;
         totalWork = beforeBatchSize * betweenBatchAndAxisSize * specIndicesSize * afterAxisSize;
     }
-
+    createOrUpdateJitKernelIfNeeded();
```

Comment:
Why the Kernel should be recreated now? It is not a shape agnostic anymore?

Reply:
Actually, the kernel was the shape agnostic only for dynamic nodes, but for nodes with static shapes its generation depends on shapes and parameters like `afterAxisSize`.